packages/core/strapi/ee/index.js

```diff
@@ -0,0 +1,168 @@
+'use strict';
+
+const { pick } = require('lodash/fp');
+
+const { readLicense, verifyLicense, fetchLicense, LicenseCheckError } = require('./license');
+const { eeStoreModel } = require('./ee-store');
+const { getRecurringCronExpression } = require('../lib/utils/cron');
+
+const ONE_MINUTE = 1000 * 60;
+const DEFAULT_FEATURES = {
+  bronze: [],
+  silver: [],
+  gold: ['sso'],
+};
+
+const ee = {
+  enabled: false,
+  licenseInfo: {},
+};
+
+const disable = (message) => {
+  ee.logger?.warn(`${message} Switching to CE.`);
+  // Only keep the license key for potential re-enabling during a later check
+  ee.licenseInfo = pick('licenseKey', ee.licenseInfo);
+  ee.enabled = false;
+};
+
+let initialized = false;
+
+const init = (licenseDir, logger) => {
+  // Can only be executed once, to prevent any abuse of the optimistic behavior
+  if (initialized) {
+    return;
+  }
+
+  initialized = true;
+  ee.logger = logger;
+
+  if (process.env.STRAPI_DISABLE_EE?.toLowerCase() === 'true') {
+    return;
+  }
+
+  try {
+    const license = process.env.STRAPI_LICENSE || readLicense(licenseDir);
+
+    if (license) {
+      ee.licenseInfo = verifyLicense(license);
+      ee.enabled = true; // Optimistically enable EE during initialization
+    }
+  } catch (error) {
+    disable(error.message);
+  }
+};
+
+const onlineUpdate = async ({ strapi }) => {
+  const transaction = await strapi.db.transaction();
```

Comment:
just a heads up the way transaction is used may be changed in the next release if the transaction PR is merged,
I'll remind my self about this if the transaction api changes

Reply:
Oh I didn't know, yes keep me posted if anything needs to be updated here 😄